File: beamforming/python/test_circular8_bf_scan.py
# SPDX-FileCopyrightText: 2021 Xianjun Jiao putaoshu@msn.com
# SPDX-License-Identifier: AGPL-3.0-or-later

import numpy as np; 
import beamforminglib as bf; 
import matplotlib.pyplot as plt

# Get the antenna response at the direction we want to scan
beam_scan_directions = np.arange(0, 360, 15)
d, wavelength = bf.ant_array_beam_pattern(freq_hz=2450e6, array_style='circular', num_ant=8, ant_spacing_wavelength=0.5, angle_vec_degree=beam_scan_directions, pause_interval=0.000001)
beamforming_vec_at_directions = -(d/wavelength)*2*np.pi
plt.close('all')

# Beam scan
for i in np.arange(len(beam_scan_directions)+1):
    bf.ant_array_beam_pattern(freq_hz=2450e6, array_style='circular', num_ant=8, ant_spacing_wavelength=0.5, beamforming_vec_rad=beamforming_vec_at_directions[i%len(beam_scan_directions),], pause_interval=0.1)
    # plt.savefig('circular8-scan-'+str(i)+'.png')

File: beamforming/matlab_octave/test_circular8_bf_scan.m
% SPDX-FileCopyrightText: 2021 Xianjun Jiao putaoshu@msn.com
% SPDX-License-Identifier: AGPL-3.0-or-later

% function test_circular8_bf_scan
clear all;
close all;

beam_scan_directions = 0:15:360;
[d, wavelength] = ant_array_beam_pattern(2450e6, 'circular', 8, 0.5, beam_scan_directions);
close all;
beamforming_vec_at_directions = -(d./wavelength).*2.*pi;

% Beam scan
for i=1:length(beam_scan_directions)
    clf;
    ant_array_beam_pattern(2450e6, 'circular', 8, 0.5, [], beamforming_vec_at_directions(i,:));
    drawnow;
end
